### ISSUE TITLE
The undocumented converse implication operator wat is not a wat

### ISSUE THREAD

**Issue by dmytrokyrychuk** [cosmologicon/pywat#18]
`**` is a [power operator](https://docs.python.org/3.5/reference/expressions.html#the-power-operator). The power operator works with numbers, so all booleans seems to be converted to integers (`True` to `1`, `False` to `0`) and then the power operation on them is calculated. Check this out:

```
>>> False ** False == True
True
```

But:

```
>>> False ** False is True
False
```

Whoopsie :) Seems like the result of the expression does not return the boolean `True`. Let's look closely:

```
>>> False ** False
1
```

Hmm. Let's check the actual type:

```
>>> type(False ** False)
<class 'int'>
```

Nope, no booleans here.

This issue needs some math to be recalled. It's commonly known that any number raised to the power of 0 is equal to 1. Also any number raised to the power of 1 is equal to itself. Let's apply this rule to the examples from [README.md](https://github.com/cosmologicon/pywat#the-undocumented-converse-implication-operator).

```
>>> False ** False
1
>>> 0 ** 0
1
```

Seems legit. 0 raised to power of 0 is equal to 1, which is also equal to `True` in Python.

```
>>> False ** True
0
>>> 0 ** 1
0
```

0 raised to power of 1 is equal to itself, which is also equal to `False` in Python. 

Two more to go:

```
>>> True ** False
1
>>> 1 ** 0
1
```

1 raised to power of 0 is equal to 1, which is also equal to `True` in Python.

```
>>> True ** True
1
>>> 1 ** 1
1
```

1 raised to power of 1 is equal to 1, which is also equal to `True` in Python.

No magic here. The power operator is yielding results, which seems to be similar to the results of the converse implication on certain subset of input values. Point me if I am wrong.

**Comment by mpanczyk** [comment]
> The power operator works with numbers, so all booleans seems to be converted to integers (True to 1, False to 0) and then the power operation on them is calculated.

Just to be clear: `bool` class inherits from `int`:

``` python
  >>> isinstance(True, int)
  True
  >>> bool.mro()
  [bool, int, object]
```

So there are available all `int` operators for `bool`.

**Comment by dmytrokyrychuk** [comment]
Agreed.

**Comment by Lucas-C** [comment]
I personally still find the analogy with the converse implication operator funny and valuable.

**Comment by ogroleg** [comment]
@Lucas-C yeah, it's funny, but not a wat

**Comment by cosmologicon** [comment]
I've added details to the explanation page.

I agree that with a lot of explanation you can understand what's going on. But I don't see why that makes it not a wat. That's essentially what a wat is, something that only makes sense with a lot of context....

**Comment by dmytrokyrychuk** [comment]
I do not agree. Power operator just raises the left operand to the rower on the right operand. It is exact purpose of the power operator. It is not a wat if you can predict the result.

There is no much context needed for this particular issue. You just need to know that bool is a subclass of int (True == 1, False == 0), which makes it possible to do any kind of operations with bools like it was int (including power).
